### DISCUSSION TITLE
Pagination

### DISCUSSION THREAD

**Discussion by RyanForward** [br-g/openf1#71] [Ideas]
Hey guys!

I really loved the API, as a fan of F1 this is a good way to match my studies with my hobby. I'm starting this discussion to introduce an idea that is common on other APIs, but I didn't found it here: pagination. 

When we're using a route to get a lot of information, it could overflow the ReadTimeout of the API, and this is sad. To avoid this, include two filters on the search could solve: page and size.

Page is the number of the page that the user is in, for example, page 1 will get the first X data from the DB.
Size is the size of data that you wanna get. Using the same example: page 1 with size 30, will get the first 30 objects from the DB.

This could improve your API use, and became this escalable, this could be a good solution for the gets of big data.

Ignore it if this resource is available, but you can answer this discussion with the location of the resource on the documentarion, cause I really did'nt found it hahah.

Thank you, guys!

**Comment by br-g** [comment]
Hello Ryan,

That sounds like a great idea!
I am currently rewriting the entire code base. The new version will make open-source collaboration much easier.
Once it's released, in a few weeks, we could plan the implementation of such features!

**Reply by RyanForward** [reply]
perfect! Thx!!
